### DISCUSSION TITLE
Enable required pin ?

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#3181] [Q&A]
Is it possible to set the vault to require the pin of the Fido2 key (webauthn) on set and login ?

if not can you move this to "feature request" ?

thanks !

**Comment by BlackDex** [comment]
See https://community.bitwarden.com/t/confusion-with-fido-u2f-and-fido2-login-and-that-fido2-pin/37249/4

Now I'm not sure if this can or should be implemented in the server side of not, if that is the case maybe it can. But else it's probably better to ask at Bitwarden regarding this i think.

**Reply by ghost** [reply]
and it's due to the lack of answer that i ask there, since it only require (when calling webauthn) to add the settings "User verification" to "preferred" (to not break legacy support of older key).

So it can be changed "technically" without any problem here (i say technically since i don't know how work the rust version of webauthn).

For me proving that you can use the key is the basic of security for a password manager and since bitwarden don't seem to care i try my luck here (but i would understand a no to stick as much as possible to them :/)

**Reply by BlackDex** [reply]
If a preferred doesn't break, it should be fine i think. Though we do want to update the library used for this on the server side, we haven't get around to it yet since there are a lot of api changes done at the library side which actually is a pain to update hehe.

**Reply by ghost** [reply]
> If a preferred doesn't break, it should be fine i think. Though we do want to update the library used for this on the server side, we haven't get around to it yet since there are a lot of api changes done at the library side which actually is a pain to update hehe.

No problem for the time, i just found sad that bitwarden does not use the "preferred" meaning if it fails the verification during the the time where the key is set, it will just work but let the user with "more secure" key use the built in webauthn user verification by pin.

edit it also mean that if the client do not support it it will technically (again don't know how work the rust implementation) will allow the login.
